PackagingProjects/CSharpPluginPayload/RepackageCSharpPlugin.ps1

```diff
@@ -0,0 +1,214 @@
+param([string]$buildConfiguration="Debug")
+
+# import GAC assembly
+Add-Type -AssemblyName "System.IO.Compression.FileSystem"
+
+if (($buildConfiguration -ne "Debug") -and ($buildConfiguration -ne "Release"))
+{
+    throw "$buildConfiguration is an invalid build configuration. Expecting Debug or Release"
+}
+
+#### Logging Helpers ############
```

Comment:
I moved all the logic inside one script because I couldn't get the import facility to work properly. The actual orchestration logic is at the bottom because all functions need to be defined first